### PR TITLE
Plasma Man tweak: NO_HUNGER + NO_EAT

### PR DESCRIPTION
**Pull request by The-Whargoul**
This PR gives plasmamen two chemical flags, NO_HUNGER and NO_EAT, The goal is to remove their requirement and ability to eat solids, as a result they can no longer take pills but also do not suffer from hunger debuffs. I made changes to pills.dm and snacks.dm to add in specific checks for plasmamen, similar to that of skellingtons. However it uses two different checks with their own in game messages. One simply checks if the plasmaman has a helmet on or not.
 
From a balance perspective. They gain the ability to work for extended periods of time without having to worry about hunger, as well as having no blood, however, they are now limited to fluids and injections for medication while still taking increased brute damage, being unable to regulate body temperature,  and are still restricted to a suit that is a massive liability.
 
tl;dr
Allows the freaks of nature to seclude themselves, as they should, and work for extended periods of time.

:cl:
 * tweak: Gave Plasmamen NO_HUNGER.
 * tweak: Removed Plasmamen's ability to eat solid food and medication.

### CODE REVIEW

**Review by The-Whargoul — COMMENTED** [review]
Oops, did not mean to commit pills and snacks.dm twice. they have been changed and the updated versions are the most recent commits

**Comment by ShiftyRail** [comment]
>commiting from Bleeding-Edge

Poor soul
I don't feel too hot about the change as it gives the plasmemes a good advantage. Not eating pills isn't that big of a disadvantage when beakers exist.

**Comment by The-Whargoul** [comment]
It is an advantage in the sense they can focus on a project. But I don't think it gives them too strong of a buff considering all of their cons. But I may be biased

**Comment by SonixApache** [comment]
What do they sustain themselves with if they don't need to eat?

**Comment by jknpj** [comment]
@SonixApache sheer anger

**Comment by SonixApache** [comment]
temperature should slowly rise throughout a round then

**Comment by The-Whargoul** [comment]
I figured the plasma inside their bio suits that is allowing them to be the walking dead to begin with.

File: code/modules/reagents/reagent_containers/food/snacks.dm [hunk 1]
```diff
@@ -218,6 +218,17 @@
 
 	if(ishuman(eater))
 		var/mob/living/carbon/human/H = eater
+		if((H.species.chem_flags & NO_EAT) && (istype(H.species,/datum/species/plasmaman)))
```
Comment (ShiftyRail) [inline code/modules/reagents/reagent_containers/food/snacks.dm]: Use `isplasmaman`

File: code/modules/reagents/reagent_containers/pill.dm [hunk 2]
```diff
@@ -42,6 +42,14 @@
 		var/mob/living/carbon/human/H = M
 		if(H.species.chem_flags & NO_EAT)
 			src.forceMove(get_turf(H))
+			if(istype(H.species,/datum/species/plasmaman))
```
Comment (ShiftyRail) [inline code/modules/reagents/reagent_containers/pill.dm]: Ditto

**Review by ShiftyRail — COMMENTED** [review]
I prefer `return FALSE` to `return 0` but I'm not sure if it matters that much

File: __DEFINES/setup.dm [hunk 3]
```diff
@@ -945,6 +945,7 @@ var/default_colour_matrix = list(1,0,0,0,\
 #define NO_SPLASH 4
 #define NO_INJECT 8
 #define NO_CRYO 16
+#define NO_HUNGER 4096
```
Comment (PJB3005) [inline __DEFINES/setup.dm]: Why is this `4096` and not `32` (next one up from `16`)?

File: code/modules/reagents/reagent_containers/food/snacks.dm [hunk 4]
```diff
@@ -218,6 +218,17 @@
...
+			if(ismob(loc))
+				var/mob/M = loc
+				M.drop_from_inventory(src)
+			src.forceMove(get_turf(H))
+		if(istype(H.head,/obj/item/clothing/head/helmet))
```
Comment (PJB3005) [inline code/modules/reagents/reagent_containers/food/snacks.dm]: Didn't you mean to indent this? This code seems broken.

Also, use `get_item_by_slot(slot_head)` instead of `H.head`

File: code/modules/reagents/reagent_containers/food/snacks.dm [hunk 5]
```diff
@@ -218,6 +218,17 @@
...
+			H.visible_message("<span class='warning'>\The [src] smashes into then slides off of the biosuit helmet, and falls to the floor below you.</span>", "<span class='notice'>\The [src] slides off of the biosuit helmet and falls to the floor below you.</span>")
```
Comment (PJB3005) [inline code/modules/reagents/reagent_containers/food/snacks.dm]: Why are you guaranteed to be wearing a biosuit helmet?

**Review by PJB3005 — CHANGES_REQUESTED** [review]

File: code/modules/reagents/reagent_containers/food/snacks.dm [hunk 6]
```diff
@@ -218,6 +218,17 @@
...
+			return
+		else
+			H.visible_message("<span class='warning'>\The [src] falls out of the putrid jaw and tumbles down onto the floor, gross. </span>", "<span class='notice'>\The [src] falls out of the putrid jaw and onto the ground below you.</span>")
+		return
```
Comment (ghost) [inline code/modules/reagents/reagent_containers/food/snacks.dm]: indent this forward a tab.

File: code/modules/reagents/reagent_containers/pill.dm [hunk 7]
```diff
@@ -42,6 +42,14 @@
...
+				if(istype(H.head,/obj/item/clothing/head/helmet))
+					H.visible_message("<span class='warning'>\The [src] slides off of the biosuit helmet, and falls to the floor.</span>", "<span class='notice'>\The [src] bounces off of the biosuit helmet, and falls to the floor</span>")
+				return 0
+			else
+				H.visible_message("<span class='warning'>\The [src] slips through a hole within the putrid jaw and falls to the floor.</span>", "<span class='notice'>You watch as \the [src] falls to the ground below you.</span>")
+			return 0
+		else
```
Comment (ghost) [inline code/modules/reagents/reagent_containers/pill.dm]: uhhhh, pretty sure you don't want that else there - can you test if humans can eat pills now

File: code/modules/reagents/reagent_containers/food/snacks.dm (hunk 4 above)
Comment (The-Whargoul) [reply code/modules/reagents/reagent_containers/food/snacks.dm]: The indention was used like that for the prior skellington NO_EAT check, It didn't cause any compiling errors/warnings so I felt it safe to leave as is.
Also tested the function locally and it works as intended.

File: code/modules/reagents/reagent_containers/pill.dm (hunk 7 above)
Comment (The-Whargoul) [reply code/modules/reagents/reagent_containers/pill.dm]: will have to test that tonight after work.

**Comment by TheFriggenSawbones** [comment]
I really don't get the point of this change, hunger was never really that much of an issue. I'd argue losing the ability to take pills is way more annoying than having to eat.

also i like being able to eat stuff

**Comment by FudgePucker** [comment]
Plasma is actually very nutritious, too bad its also extremely toxic.

File: __DEFINES/setup.dm (hunk 3 above)
Comment (The-Whargoul) [reply __DEFINES/setup.dm]: Fixed

**Comment by The-Whargoul** [comment]
@TheFriggenSawbones the point is to allow the species to work on lengthy projects without being interrupted every few minutes. With all of the debuffs plasmamen have, I figured this would give them something more to work with, as they used to not have to eat in the past, and removing the hunger debuff w/ it's speed reduction(that is hell when combined with their normal speed reduction from their suits) will benefit plasmamen engineers/miners/and scientists significantly with their relative jobs.
I didn't want to try to give a pro without giving a new con, so that's why NO_EAT was added. It's not too difficult to work around, plus a species, that is unable to remove their suit without bursting into flames, wouldn't be able to eat solid foods. But with that being said, realism doesn't quite matter.

**Comment by Coxswain-Navigator** [comment]
I think this is a very good change, as it makes sense logically and even in terms of gameplay. Even if being unable to comsume pills may be a challenge for some players, a bit of creativity is enough to overcome it. This is a huge buff for any jobs that spend a lot of time away from the station such as spacepod mechanics (assuming they explore space) and asteroid-related jobs.

**Comment by Blithering** [comment]
Plasmamen are already the worst race to have turn up in medbay, they can't use bandages, ointments or the advanced kits, they can barely use cryo thanks to permeability code, if they were exposed to air for any time they'll be boiling in their suits and cloning them is a scramble to get them suited before they combust. Adding no pills on top of this is a bad idea in my opinion. Plus the upside they get as a tradeoff will just make chefs less useful and people more likely to squirrel away for two hours making something rather than interacting.

File: code/modules/reagents/reagent_containers/pill.dm [hunk 8]
```diff
@@ -42,8 +42,14 @@
 		var/mob/living/carbon/human/H = M
 		if(H.species.chem_flags & NO_EAT)
 			src.forceMove(get_turf(H))
-			H.visible_message("<span class='warning'>\The [src] falls through and onto the ground.</span>", "<span class='notice'>You hear \the [src] plinking around for a second before it hits the ground below you.</span>")
-			return 0
+			if(isplasmaman(H))
+				if(H.is_wearing_item(/obj/item/clothing/head/helmet,slot_head))
+					H.visible_message("<span class='warning'>\The [src] slides off of the sealed helmet, and falls to the floor.</span>", "<span class='notice'>\The [src] bounces off of the sealed helmet, and falls to the floor</span>")
```
Comment (ShiftyRail) [inline code/modules/reagents/reagent_containers/pill.dm]: Forgot a dot.

Comment (The-Whargoul) [reply code/modules/reagents/reagent_containers/pill.dm]: fixed.

**Comment by The-Whargoul** [comment]
@Blithering I'd of liked to not take more away from the species, however, for consistency and balance reasons. I felt removing their ability to take pills was reasonable. I don't think that is more of a detriment to the doctors more so than it is to the plasmaman player. But simply popping a pill into a beaker or bottle or any container of the sort will easily solve that issue.
With that being said, if you were to have to take something away to give them this QoL buff, what would you suggest?

**Comment by ghost** [comment]
All aliens at current are just humans with a gimmick, not even good ones. Vox need internals constantly and grays are just humans with telepathy. This change will make plasmamen really alien and hardly similar to humans at all.
Only real argument against it is that it'd make playing plasmamen harder, which is a good thing. If every race places virtually identically to humans, whats the point in aliens? If you say because they're cool and give more customisation options, I look forward to unathi, tajaran and skrell round start.

**Comment by Irradiation** [comment]
While you're at it, make it so ingesting plasma doesn't give toxins to Plasmamemes. I always found it weird that the magical substance that they breathe doesn't do anything bad to them, but DRINKING it gives them the same toxin effect as any other race ingesting plasma would have.

**Review by PJB3005 — APPROVED** [review]

**Comment by ghost** [comment]
While I don't have a particular investment in plasmamen as a race, I am heavily in favor of a greater distinction between races.  Especially ones that give races a less human feel.  Anthropomorphism in the spess genre is easily one of my biggest pet peeves (looking at you, Virgo).  While it could be argued from a lore standpoint that they were once human, plasma is a crazy enough thing to heavily modify people into being not-people (example being the application of plasma to black slime extract to assist in the creation of adv slime tox).  It baffles me when people make bottles of meds for situations like this when you can literally toss any pill into a space cola can to drink it  (as WELL as screen the reagents).  In the case of docs, a pocket beaker can get any pill ready for this in no time.

**Comment by dickmuncher** [comment]
Sawbones is against it yet he doesn't play the game anymore. So I approve of this change. Remember when Plasmamen used to scream everything they said in caps? What happened to that, too bad sawbones doesn't really care about Plasmamen or he'd realize that was the one thing that made them unique.

Thanks fagbones, I give my +1 to this update

**Comment by SonixApache** [comment]
ah yes, let us make plasmemes more like skelememes
that will make them more alien and less human

**Comment by The-Whargoul** [comment]
@SonixApache they'd share two flags rather than one. No_Blood and No_Eat. I wouldn't say this is trying to make plasmamen more like skellingtons. With this change, plasmamen would be alone in having No_Hunger.

**Comment by Blithering** [comment]
@the-whargoul personally I'd make them come out of cloners as humans. Makes sense lore and gameplay wise, and makes plasmamen actually horribly scarred humans on super poison life support rather than a proper race. That way you get one life as a burning skelebones and have to be careful if you want to stay as one. It would also help the doctors by bypassing the usual plasmaman post cloning issues

**Comment by ghost** [comment]
actually that's a great idea @Blithering

**Comment by ghost** [comment]
@Blithering well I coded it but I did a provisional vote in game and it wasn't a particularly popular feature idea

**Comment by Hugh-G-Rection** [comment]
I feel it would make more loresense to make them uncloneable. Should only have the option to just stuff their brain in a monkeyman, that's what I like to do anyways.

**Comment by The-Whargoul** [comment]
@Hugh-G-Rection  I'm fine with making them hard mode, but not with this proposed change.

**Comment by SerWind** [comment]
I haven't read all the comments here. I imagine there's a lot of good points. I've read some of them though, and I like (as has been mentioned) anything that differentiates the races, but playing non human is always suppose to be a negative (I think?) so I would prefer it be a negative that makes them unique. 

Obviously it's a bit give and take, but apparently they can still use beakers? I think the whole concept of them not eating and sustaining themselves on plasma really works. Like how do they even eat. It helps with the immulsions. But I'd prefer to avoid an era of Sawbone esque power gamers rolling assistant for 100 rounds to get antag as a plasma man. Even though I've never been a fan of "bad people will do bad things with it" mentality.

Maybe a little tweaking is in order? Otherwise I support it, for what it's worth

**Comment by Blithering** [comment]
I really dislike the concept that non human races need to be horrifically crippled. As it stands I can't recall anyone who was banned for excessive powergaming that didn't play as a human to do so, with Asimov being as powerful as it is. I guess people will never stop thinking Sawbones when they read plasmaman, and so you'll always get the people like dickmuncher voting against whatever benefits him

**Comment by SonixApache** [comment]
skelememes were extremely powergamey until they got deleted and nerfed to shit
dionae were extremely powergamey until they got nerfed to shit
i honestly don't know about plasmemes, but i assume they were nerfed to shit along the way as well

humans are the baseline, other races are gimmicks

**Comment by Blithering** [comment]
As far a I know, plasmamen were only powergamey at two points, when their helmets (and vox helmets) had flash protection, and when a bug doubled their burn resistance. They're still riddled with bugs like being unable to cool down (thankfully a PR is open to fix that after a year) and spawning with emergency O2 tanks, and I feel they really don't need to be made more of a pain to play as one for the sake of some mild convenience like no hunger. Gimmick or no, no race should be more of an arse to play than any other with the definite exception of catbeasts

**Comment by StreptoMutans** [comment]
This killed feeding pills and the hunger system for everyone

**Comment by Kammerjunk** [comment]
Impressive
Very nice

**Comment by Blithering** [comment]
2016-04-15-beartato-fixcomputer

**Comment by The-Whargoul** [comment]
Right, seems at one point or another indentation got fucked up in handle_regular_hud_updates, which gave everyone the nutrition0 icon regardless of nutrition level, my guess would be due to github editor.
Should be able to fix tonight and provide a more thorough test. 

OOPS

**Comment by SonixApache** [comment]
stop using DM you absolute buffoon

**Comment by The-Whargoul** [comment]
OK MOM!
But ye, I know.
